packages/docs/src/components/MedplumCodeBlock.tsx

```diff
@@ -0,0 +1,103 @@
+import CodeBlock, { Props } from '@theme/CodeBlock';
+import * as React from 'react';
+
+const BLOCK_START_PATTERN = new RegExp('^//\\s*start-block\\s+([A-Za-z_-]*)', '');
+// const BLOCK_START_PATTERN = /^\/\/\s*start-block\s+([A-Za-z_-]*)/;
+const BLOCK_END_PATTERN = new RegExp('^//\\s*end-block\\s+([A-Za-z_-]*)', '');
+
+interface MedplumCodeBlockProps extends Props {
+  selectLines?: number[][];
+  selectBlocks?: string[] | string;
+}
+
+export default function MedplumCodeBlock({
+  children,
+  selectLines,
+  selectBlocks,
+  ...props
+}: MedplumCodeBlockProps): JSX.Element {
+  let code = children as string;
+
+  if (selectLines && selectBlocks) {
+    throw new Error("Only one of 'selectLines' or 'selectBlocks' can be specified");
+  }
+
+  let codeLines: (string | null)[] = code.split('\n');
```

Comment:
I assume `code` is required?  Then the cast and `| null` should not be required.